### ISSUE TITLE
Bundle whole set of icons

### ISSUE THREAD

**Issue by acalvino4** [antfu/unplugin-icons#186]
Can this plugin be used to bundle a whole icon set?

The use case is user-generated content: I want to be able to specify an icon set (or sets) from my build process or from my JS, then have the content author be able to select an icon from that set via my CMS backend. I can't know ahead of time which icon will be selected - it is simply specified in my backend template as the content of a field, i.e. `{{ post.svgField }}` where that template tag outputs a link to the processed svg file under `dist`.

Would this work?
```js
// main.js

import * from '~icons/carbon
```

Or would I have to actually use the icon in my js for it to be bundled? And would I have to be using React or Vue?

Basically I just want a straightforward way to have vite dump a specified set of optimized svgs into `dist/svg/setName`.

If this is not possible, consider this a feature request!

**Comment by airtonix** [comment]
related #190

**Comment by userquin** [comment]
you can use this simple icon for each library, just change the static import (this is an example using vue, port it to your fw):
**note**: you can also provide attributes overriding them (width, height...), keep in mind the full icon list will go to the sfc asset, you can use a vite virtual plugin to remove part of the icons, but you should do it.
```ts
<script setup lang="ts">
import { computed } from 'vue'
import { icons } from '@iconify-json/carbon'

const props = defineProps<{ icon: string }>()

const meta = computed(() => {
  const { left = 0, top = 0, width = '1em', height = '1em' } = icons ?? {}

  return { left, top, width, height }
})

const viewBox = computed(() => {
  return `${meta.value.left} ${meta.value.top} ${meta.value.width} ${meta.value.height}`
})

const svgIcon = computed(() => {
  return icons.icons[props.icon]
})
</script>

<template>
  <svg
    v-if="svgIcon"
    :width="meta.width"
    :height="meta.height"
    :viewBox="viewBox"
    xmlns="http://www.w3.org/2000/svg"
    v-html="svgIcon.body"
  />
</template>
```

**Comment by antfu** [comment]
Close as out-of-scope. I would suggest using Iconify's runtime solution if your usage is dynamic.
